app/controllers/admin/comments_controller.rb

```diff
@@ -17,16 +19,46 @@ def index
                       .order(created_at: :desc)
                       .page(params[:page] || 1).per(50)
                   end
+      @countable_vomits = {}
+      @comments.each do |comment|
+        @countable_vomits[comment.id] = calculate_flags_for_single_comment(comment)
+      end
     end
 
     def show
       @comment = Comment.includes(:user, :commentable).find(params[:id])
+      @countable_vomits = {}
+      @countable_vomits[@comment.id] = calculate_flags_for_single_comment(@comment)
     end
 
     private
 
     def authorize_admin
       authorize Comment, :access?, policy_class: InternalPolicy
     end
+
+    def calculate_flags_for_single_comment(comment)
+      privileged_comment_reactions = comment.reactions.privileged_category.select do |reaction|
+        reaction.reactable_type == "Comment"
+      end
+      vomit_comment_reactions = privileged_comment_reactions.select { |reaction| reaction.category == "vomit" }
+      vomit_count = 0
+
+      if vomit_comment_reactions.present?
+        vomit_comment_reactions.each do |reaction|
+          vomit_count += 1 if reaction.status != "invalid"
+        end
+      end
+
+      vomit_count
+    end
```

Comment:
I am not sure why we are checking that `reaction.reactable_type == "Comment"`, if I understand the [`Reactable`](app/models/concerns/reactable.rb) module correctly then `comment.reactions` will always only return reactions with a reactable_type of `"Comment"`

But also since we are not using the intermediate results of the filtering outside this method, I think we can reduce counting the vomits to just this:

```suggestion
    def calculate_flags_for_single_comment(comment)
      comment.reactions.privileged_category.select do |reaction|
        reaction.category == "vomit" && reaction.status != "invalid"
      end.count
    end
```

Reply:
Thanks a lot for code optimisation and clarification on Reactable.

I have updated code to:

```
    def calculate_flags_for_single_comment(comment)
      comment.reactions.privileged_category.count do |reaction|
        reaction.category == "vomit" && reaction.status != "invalid"
      end
    end
```